packages/components/src/color-picker/index.js

```diff
@@ -43,30 +43,144 @@ import Alpha from './alpha';
+import { colorToState, simpleCheckForValidColor, isValidHex } from './utils';
+
+const toLowerCase = ( value ) => String( value ).toLowerCase();
+const isValueEmpty = ( data ) => ( data.source === 'hex' && ! data.hex ) ||
+	( data.source === 'hsl' && ( ! data.h || ! data.s || ! data.l ) ) ||
+	( data.source === 'rgb' && (
+		( ! data.r || ! data.g || ! data.b ) &&
+		( ! data.h || ! data.s || ! data.v || ! data.a ) &&
+		( ! data.h || ! data.s || ! data.l || ! data.a )
+	) );
+const isValidColor = ( colors ) => colors.hex ?
+	isValidHex( colors.hex ) :
+	simpleCheckForValidColor( colors );
+
+/**
+ * Function that creates the new color object
+ * from old data and the new value.
+ *
+ * @param {Object} oldColors The old color object.
+ * @param {string} oldColors.hex
+ * @param {Object} oldColors.rgb
+ * @param {number} oldColors.rgb.r
+ * @param {number} oldColors.rgb.g
+ * @param {number} oldColors.rgb.b
+ * @param {number} oldColors.rgb.a
+ * @param {Object} oldColors.hsl
+ * @param {number} oldColors.hsl.h
+ * @param {number} oldColors.hsl.s
+ * @param {number} oldColors.hsl.l
+ * @param {number} oldColors.hsl.a
+ * @param {string} oldColors.draftHex Same format as oldColors.hex
+ * @param {Object} oldColors.draftRgb Same format as oldColors.rgb
+ * @param {Object} oldColors.draftHsl Same format as oldColors.hsl
+ * @param {Object} data Data containing the new value to update.
+ * @param {Object} data.source One of `hex`, `rgb`, `hsl`.
+ * @param {string\number} data.value Value to update.
+ * @param {string} data.valueKey Depends on `data.source` values:
+ *   - when source = `rgb`, valuKey can be `r`, `g`, `b`, or `a`.
+ *   - when source = `hsl`, valuKey can be `h`, `s`, `l`, or `a`.
+ * @return {Object} A new color object for a specific source. For example:
+ * { source: 'rgb', r: 1, g: 2, b:3, a:0 }
+ */
+const dataToColors = ( oldColors, { source, valueKey, value } ) => {
+	if ( source === 'hex' ) {
+		return {
+			source,
+			[ source ]: value,
+		};
+	}
+	return {
+		source,
+		...{ ...oldColors[ source ], ...{ [ valueKey ]: value } },
+	};
+};
 
 export default class ColorPicker extends Component {
 	constructor( { color = '0071a1' } ) {
 		super( ...arguments );
-		this.state = colorToState( color );
-		this.handleChange = this.handleChange.bind( this );
+		const colors = colorToState( color );
+		this.state = {
+			...colors,
+			draftHex: toLowerCase( colors.hex ),
+			draftRgb: colors.rgb,
+			draftHsl: colors.hsl,
+		};
+		this.commitValues = this.commitValues.bind( this );
+		this.setDraftValues = this.setDraftValues.bind( this );
+		this.resetDraftValues = this.resetDraftValues.bind( this );
+		this.handleInputChange = this.handleInputChange.bind( this );
 	}
 
-	handleChange( data ) {
+	commitValues( data ) {
 		const { oldHue, onChangeComplete = noop } = this.props;
-		const isValidColor = simpleCheckForValidColor( data );
-		if ( isValidColor ) {
+
+		if ( isValueEmpty( data ) ) {
+			return;
+		}
```

Comment:
No, it's because of the former `handleChange` method in the inputs component, see https://github.com/WordPress/gutenberg/pull/11772/files#diff-2a6da6e04df0dda01acf1714b4d7d4b0L120 Note how the if statements checked for non-empty values as in `if ( data.hex )`.

The former HEX case you mention was also lifted up and is now part of the `isValidColor` check.

Reply:
Sounds good.